File: codes/scala/chapter_stack_and_queue/queue.scala
import scala.collection.mutable

/**
 * File: queue.scala
 * Author: WangBlue (wangjiping596@gmail.com)
 * Create: 2024-08-11 11:50
 */


object queue {
  def main(args: Array[String]): Unit = {
    val queue: mutable.Queue[Int] = new mutable.Queue[Int]()

    /* 元素入队 */
    queue.enqueue(1)
    queue.enqueue(3)
    queue.enqueue(2)
    queue.enqueue(5)
    queue.enqueue(4)


    // 打印队列
    println(s"队列 queue = $queue")

    // 访问队首元素
    val peek: Int = queue.headOption.getOrElse(-1)
    println(s"队首元素 peek = $peek")

    // 元素出队
    val pop: Int = if (queue.nonEmpty) queue.dequeue() else -1
    println(s"出队元素 pop = $pop，出队后 queue = $queue")

    // 获取队列的长度
    val size: Int = queue.size
    println(s"队列长度 size = $size")

    // 判断队列是否为空
    val isEmpty: Boolean = queue.isEmpty
    println(s"队列是否为空 = $isEmpty")
  }

}

File: ru/codes/csharp/chapter_stack_and_queue/queue.cs
/**
 * File: queue.cs
 * Created Time: 2022-12-23
 * Author: haptear (haptear@hotmail.com)
 */

namespace hello_algo.chapter_stack_and_queue;

public class queue {
    [Test]
    public void Test() {
        /* Инициализировать очередь */
        Queue<int> queue = new();

        /* Поместить элемент в очередь */
        queue.Enqueue(1);
        queue.Enqueue(3);
        queue.Enqueue(2);
        queue.Enqueue(5);
        queue.Enqueue(4);
        Console.WriteLine("Очередь queue = " + string.Join(",", queue));

        /* Получить элемент в начале очереди */
        int peek = queue.Peek();
        Console.WriteLine("голова очередиэлемент peek =" + peek);

        /* Извлечь элемент из очереди */
        int pop = queue.Dequeue();
        Console.WriteLine("Элемент, извлеченный из очереди, pop = " + pop + ", queue после извлечения = " + string.Join(",", queue));

        /* Получить длину очереди */
        int size = queue.Count;
        Console.WriteLine("Длина очереди size =" + size);

        /* Проверить, пуста ли очередь */
        bool isEmpty = queue.Count == 0;
        Console.WriteLine("Очередь пуста: " + isEmpty);
    }
}
